### ISSUE TITLE
SH-2 compiler

### ISSUE THREAD

**Issue by smdstudios** [fadedled/seta-gx#2]
Picodrive on Retroarch Wii has full-speed 32X emulation in all the games I've tested. I wonder if there's anything in Picodrive's SH-2 compiler that could provide insights for Seta GX's development?

**Comment by fadedled** [comment]
Yes, I've looked at Picodrive's SH2 dynarec, the plan is to make a dynarec from scratch but it helps to see how things might be implemented :) I've also looked at many other recompilers.

**Comment by fadedled** [comment]
Hey! the Dreamcast is very different from the Saturn, although they do share the same CPU family, so I don't think it will be of much use hehe. For the next release the big thing will be a new SH2 dynarec, right now it's mostly complete and I have it working in basic tests, right now I'm focusing on setting up the code to be able to switch from interpreter to dynarec and test it in real conditions (If everything goes according to the tests, it would add a general 2-3x speed improvement).

**Comment by satmonk** [comment]
You have lost a month of work? A Harddrive-failure? Maybe I can help?

**Comment by jilou08** [comment]
Hello ^^,

I read this, I don't know if it's interesting/helpful or not... It's about the sound in Sega Saturn:

https://www.segasaturnshiro.com/2026/03/16/frogbull-breakthrough-demo-paves-the-way-for-sequenced-music-in-homebrew-saturn-games/


Have a good day ^^.

**Comment by jilou08** [comment]
Hello,

I have a question: could the difficulties in getting some games to run with the new dynarec be caused by an issue introduced during the simplification of Yabause?

I noticed that several games such as Shining Wisdom, Shining the Holy Ark, Sonic Jam, Sonic Xtreme, Sonic Ztreme, Keio Squadron 2, etc. do not work with SetaGX, while they run (with no graphic bug but slowly) with Yabause on Wii.

(I think about that today and I try these games on Yabause today too...) 

Shining the Holy Ark and Shining Wisdom on Yabause:

<img width="4000" height="3000" alt="Image" src="https://github.com/user-attachments/assets/5c381e79-e1db-48b6-b09f-2d50a0a9022b" />

<img width="4000" height="3000" alt="Image" src="https://github.com/user-attachments/assets/23b2fa89-b3fb-4ae0-b035-01b7fe137561" />


Have a good day ^^.

**Comment by fadedled** [comment]
Hello, I can confirm that all changes made to the emulator work on the previous interpreter the same, the main changes that make those games work differently are probably because of changes made to the SMPC code. Right now I'm trying to match the dynarec and the previous interpreter, they're almost 1:1 but there is one small bug that I'm not seeing that makes games not boot (Only Megaman 8 boots but crashes eventually).

**Comment by jilou08** [comment]
Hello ^^,

I really admire the work you're doing on this emulator.
Hopefully that small bug will be resolved soon, it really feels like you're very close now. Looking forward to the update.

Also, it's great to see you adding new controller support to the emulator. Keep up the excellent work.

Have a good day ^^.
